### PR TITLE
[SPARK-36689][PYTHON] Cleanup the deprecated APIs and raise proper warning message.

### PR DESCRIPTION
**Pull request by itholic**
### What changes were proposed in this pull request?

This PR proposes cleanup the deprecated APIs in `missing/*.py`, and raise proper warning message for the deprecated APIs such as pandas does.

Also remove the checking for pandas < 1.0, since now we only focus on following the behavior of latest pandas.

### Why are the changes needed?

We should follow the deprecation of APIs of latest pandas.

### Does this PR introduce _any_ user-facing change?

Now the some APIs raise proper alternative message for deprecated functions such as pandas does.

### How was this patch tested?

Ran `dev/lint-python` and manually check the pandas API documents one by one.

### CODE REVIEW

**Review by HyukjinKwon — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Can you fix tests?

```
======================================================================
ERROR [0.038s]: test_missing (pyspark.pandas.tests.test_series.SeriesTest)
----------------------------------------------------------------------
pyspark.pandas.exceptions.PandasNotImplementedError: The method `pd.Series.slice_shift()` is deprecated in pandas and will therefore not be supported in pandas-on-Spark. Use `shift` instead.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/__w/spark/spark/python/pyspark/pandas/tests/test_series.py", line 1040, in test_missing
    getattr(psser, name)()
AssertionError: "method.*Series.*tshift.*is deprecated" does not match "The method `pd.Series.slice_shift()` is deprecated in pandas and will therefore not be supported in pandas-on-Spark. Use `shift` instead."

----------------------------------------------------------------------
```

Otherwise looks good. cc @ueshin too fyi

File: python/pyspark/pandas/missing/series.py [hunk 1]
```diff
@@ -47,22 +43,19 @@ class MissingPandasLikeSeries(object):
...
     # Deprecated functions
-    convert_objects = _unsupported_function("convert_objects", deprecated=True)
-    nonzero = _unsupported_function("nonzero", deprecated=True)
-    reindex_axis = _unsupported_function("reindex_axis", deprecated=True)
-    select = _unsupported_function("select", deprecated=True)
-    get_values = _unsupported_function("get_values", deprecated=True)
+    slice_shift = _unsupported_function(
+        "slice_shift", deprecated=True, reason="Use DataFrame/Series.shift instead."
+    )
+    tshift = _unsupported_function("slice_shift", deprecated=True, reason="Use `shift` instead.")
```
Comment (ueshin) [inline python/pyspark/pandas/missing/series.py:58]: `"tshift"` instead of `"slice_shift"`?

**Review by ueshin — APPROVED** [review]
Otherwise, LGTM, pending tests.

File: python/pyspark/pandas/missing/series.py (hunk 1 above)
Comment (itholic) [reply python/pyspark/pandas/missing/series.py]: Thanks!

**Comment by HyukjinKwon** [comment]
Merged to master.

**Pull request closed without merging**